pkg/policy/match.go

```diff
@@ -20,6 +20,20 @@ func (p Policy) Match(node datamodel.Node) bool {
 	return true
 }
 
+// Filter performs a recursive filtering of the Statement, and prunes what doesn't match the given path
+func (p Policy) Filter(path ...string) Policy {
```

Comment:
Since the entire policy is IPLD, why not use https://pkg.go.dev/github.com/ipld/go-ipld-prime?utm_source=godoc#Path for the `path` argument.

Reply:
The policy is IPLD, but the semantic here is fairly different than an IPLD path. I think it'd be just more confusing.

Let's try like that and see if problems surface.

Reply:
> it might also be possible to use the [traversal](https://pkg.go.dev/github.com/ipld/go-ipld-prime@v0.21.0/traversal) package to find the desired subset of nodes and then create the policy object from that subset

it's not, we don't filter the graph of policies, we filter the selector and what is being attached to them